huaweicloud/services/dds/resource_huaweicloud_dds_instance_parameters_modify.go

```diff
@@ -0,0 +1,182 @@
+package dds
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// @API DDS PUT /v3/{project_id}/instances/{instance_id}/configurations
+// @API DDS GET /v3/{project_id}/instances
+// @API DDS GET /v3/{project_id}/jobs
+func ResourceDDSInstanceParametersModify() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDDSInstanceParametersModifyCreate,
+		ReadContext:   resourceDDSInstanceParametersModifyRead,
+		DeleteContext: resourceDDSInstanceParametersModifyDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"instance_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"parameters": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+						"value": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+			"entity_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceDDSInstanceParametersModifyCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	client, err := conf.DdsV3Client(conf.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating DDS client: %s ", err)
+	}
+	instId := d.Get("instance_id").(string)
+	entityId := d.Get("entity_id").(string)
+	if entityId == "" {
+		entityId = instId
+	}
+	opt := buildDDSInstanceParametersModifyRequestBody(d.Get("parameters").([]interface{}), entityId)
+
+	// modify parameters
+	if ctx, err = modifyInstanceParameters(ctx, client, d.Timeout(schema.TimeoutCreate), instId, opt); err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(instId)
```

Comment:
the ID is error, if group or node params are modified, then the instance ID may same, but entity ID is different, so the resource ID should contains entity ID if the valuie of entity_id is not empty